src/java/org/apache/cassandra/index/sai/QueryContext.java

```diff
@@ -254,14 +244,10 @@ public void updateAnnRerankFloor(float observedFloor)
             annRerankFloor = max(annRerankFloor, observedFloor);
     }
 
-    public void setOriginalPlan(Plan originalPlan)
+    public void recordQueryPlan(Plan.RowsIteration originalPlan, Plan.RowsIteration.RowsIteration optimizedPlan)
```

Comment:
Should be: `public void recordQueryPlan(Plan.RowsIteration originalPlan, Plan.RowsIteration optimizedPlan)`